DeepCat.js

```diff
@@ -209,7 +208,24 @@
 		newSearchTerms = DeepCat.computeResponses( responses, newSearchTerms );
 		newSearchTerms = DeepCat.computeErrors( errors, newSearchTerms );
 
-		substituteSearchRequest( newSearchTerms.join( ' ' ) );
+		logAndFinishRequest( newSearchTerms.join( ' ' ) );
+	}
+
+	function logAndFinishRequest( newSearchTermString ) {
+		var logRequestUrl = '//tools.wmflabs.org/catgraph-jsonp/logrequestlength?searchquerylength={0}';
+		return $.ajax( {
+			url: stringFormat( logRequestUrl, newSearchTermString.length ),
+			timeout: ajaxTimeout,
+			cache: false,
+			complete: function() {
+				finishDeepCatRequest( newSearchTermString );
+			}
+		} );
+	}
+
+	function finishDeepCatRequest( newSearchTermString ) {
+		substituteSearchRequest( newSearchTermString );
+		removeAjaxThrobber();
```

Comment:
In a first version I tried to use a fire and forget mechanism but it lead to some strange behaviour sometimes not triggering the logging on the server at all. So I changed that.

We cant do the logging on the server-side for two reasons:

 - When the user uses more then one keyword in a search the server gets several requests. Atm the server does not now which requests belong together and which do not. True we could solve this by transmitting an unique request identifier. 

- The server does not now the whole request. Meaning the non-deepcat related part. E.g.: "deepcat:Musik Lorem ipsum dolor ......"